File: algorithms/3289.the-two-sneaky-numbers-of-digitville/solution.cpp
#include <unordered_map>
#include <vector>

using namespace std;

class Solution {
public:
    vector<int> getSneakyNumbers(vector<int>& nums) {
        vector<int> sneakyNumbers;
        unordered_map<int, int> count;
        for (int num : nums) {
            count[num]++;
            if (count[num] > 1) {
                sneakyNumbers.push_back(num);
            }
            if (sneakyNumbers.size() == 2) {
                break;
            }
        }
        return sneakyNumbers;
    }
};

File: algorithms/3289.the-two-sneaky-numbers-of-digitville/solution.rs
use std::collections::HashMap;

impl Solution {
    pub fn get_sneaky_numbers(nums: Vec<i32>) -> Vec<i32> {
        let mut sneaky_numbers = Vec::new();
        let mut count = HashMap::new();
        for num in nums {
            *count.entry(num).or_insert(0) += 1;
            if count[&num] > 1 {
                sneaky_numbers.push(num);
            }
            if sneaky_numbers.len() == 2 {
                break;
            }
        }
        sneaky_numbers
}
